pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +120,325 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+			if _, err := client.Default().Dynamic().Resource(gvrNetnamespace).Update(ctx, nns, metav1.UpdateOptions{}); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil // success
+}
+
+func convertOvnEgressIpToSdnEgressIp(ctx context.Context, client cnoclient.Client) error {
+	egressIpRollbackReady, err := sdnEgressIpResourcesReady(ctx, client)
+	if !egressIpRollbackReady {
+		return nil // wait for all SDN egressIP resources to be created before rollback
+	} else if err != nil {
+		return err
+	}
+
+	// 1. query for egressips
+	egressIpList, err := ListAllOfSpecifiedType(gvrEgressIp, ctx, client)
+	if err != nil {
+		return err
+	}
+
+	// 2. query for nodes
+	nodeList, err := ListAllNodes(ctx, client)
+	if err != nil {
+		return err
+	}
+
+	// 3. iterate through egressIP objects and generate netnamespaces
+	//    - value of `namespaceSelector` will determine netnamespace's name
+	//    - egressIP list will be copied to egressIPs field for netnamespace
+	for _, eip := range egressIpList {
+		// translate egressIPs to SDN netnamespaces
+		// - delete cloudprivateipconfig if it exists
+		// - create egressip to netnamespace resources for sdn
+
+		egressIps := eip.Object["spec"].(map[string]interface{})["egressIPs"].([]interface{})
+
+		if err := deleteCloudPrivateIpConfigs(ctx, client, egressIps); err != nil {
+			return err
+		}
+
+		eipNamespace := eip.Object["spec"].(map[string]interface{})["namespaceSelector"].(map[string]interface{})["matchLabels"].(map[string]interface{})["kubernetes.io/metadata.name"]
+		eipNamespaceStr := fmt.Sprintf("%v", eipNamespace)
+		nns, err := client.Default().Dynamic().Resource(gvrNetnamespace).Get(ctx, eipNamespaceStr, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		nns.Object["egressIPs"] = egressIps
+		if err := apply.ApplyObject(ctx, client, nns, ""); err != nil {
+			return err
+		}
+	}
+
+	// 4. iterate through nodes and generate hostsubnets for any that are egress-assignable
+	for _, node := range nodeList {
+		if _, ok := node.Labels["k8s.ovn.org/egress-assignable"]; ok {
+			// generate egressCIDRs field
+			// if "sdnEgressCIDRs" node label exists, automatic sdn config was used and we shall reuse old config
+			// else, manual sdn config was used and we shall use node's subnet CIDR (cannot restore egressIPs)
+			var egressCIDRsArr []string
+			if _, ok := node.Annotations["sdn.to.ovn.migration/sdnEgressCIDRs"]; ok {
+				if err := json.Unmarshal([]byte(node.Annotations["sdn.to.ovn.migration/sdnEgressCIDRs"]), &egressCIDRsArr); err != nil {
+					return err
+				}
+			} else {
+				egressIpConfig := node.Annotations["cloud.network.openshift.io/egress-ipconfig"]
+
+				r := bytes.NewReader([]byte(egressIpConfig))
+				decoder := json.NewDecoder(r)
+
+				egressIpConfigArr := make([]NodeEgressIpConfig, 0)
+				if err := decoder.Decode(&egressIpConfigArr); err != nil {
+					return err
+				}
+				nodeSubnet := egressIpConfigArr[0].IfAddr["ipv4"]
+				egressCIDRsArr = []string{nodeSubnet}
```

Comment:
I wonder way a standard un-marshaling could not be done here?

Reply:
I found this approach to be elegant and it does not require any additional dependencies (all are from standard library).

Reply:
But I was wondering why this wouldn't work (I don't see additional dependencies here)
```
if err := json.Unmarshal([]byte(egressIpConfig), &egressIpConfigArr); err != nil {
    return err
}
```
instead of
```
r := bytes.NewReader([]byte(egressIpConfig))
decoder := json.NewDecoder(r)

egressIpConfigArr := make([]NodeEgressIpConfig, 0)
if err := decoder.Decode(&egressIpConfigArr); err != nil {
    return err
}
```

Reply:
Yes standard un-marshaling works fine. I guess since my code is to some degree confusing I'll switch to standard un-marshaling instead. 

Like:
```
egressIpConfig := node.Annotations[egressIPNodeConfig]
egressIpConfigArr := make([]NodeEgressIpConfig, 0)
if err := json.Unmarshal([]byte(egressIpConfig), &egressIpConfigArr); err != nil {
	return err
}
nodeSubnet := egressIpConfigArr[0].IfAddr["ipv4"]
egressCIDRsArr = []string{nodeSubnet}